### ISSUE TITLE
A uncaught exception occured: TypeError: resolver is not a function

### ISSUE THREAD

**Issue by NemoN** [sirchia/pimatic-rflink#14]
Got this error some days ago. It is possible to catch this exception?

```
error [pimatic]: A uncaught exception occured: TypeError: resolver is not a function
    at Board._handleAcknowledge (/root/thomas/pimatic-app/node_modules/pimatic-rflink/lib/board.coffee:183:5)
    at Board._processData (/root/thomas/pimatic-app/node_modules/pimatic-rflink/lib/board.coffee:128:38)
    at SerialPortDriver.<anonymous> (/root/thomas/pimatic-app/node_modules/pimatic-rflink/lib/board.coffee:43:8)
    at emitOne (events.js:77:13)
    at SerialPortDriver.emit (events.js:169:7)
    at SerialPort.<anonymous> (/root/thomas/pimatic-app/node_modules/pimatic-rflink/lib/serialport.coffee:37:8)
    at emitOne (events.js:77:13)
    at SerialPort.emit (events.js:169:7)
    at /root/thomas/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/parsers.js:25:17
    at Array.forEach (native)
    at Object.parser (/root/thomas/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/parsers.js:24:13)
    at Object.SerialPort.opts.dataCallback (/root/thomas/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/serialport.js:181:12)
    at SerialPortFactory.SerialPort._emitData (/root/thomas/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/serialport.js:396:20)
    at afterRead (/root/thomas/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/serialport.js:372:20)
    at /root/thomas/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/serialport.js:388:9
    at FSReqWrap.wrapper [as oncomplete] (fs.js:576:17)
 This is most probably a bug in pimatic or in a module, please report it!
```

**Comment by nanokatz** [comment]
I am also having this issue on a new install. 

> 15:23:12.312 2016-10-14 Friday
> 15:23:23.364 [pimatic] Starting pimatic version 0.9.34
> 15:23:28.278 [pimatic, ppm] 2.15.6
> 15:23:28.373 [pimatic] Loading plugin: "pimatic-cron" (0.8.8)
> 15:23:29.545 [pimatic] Loading plugin: "pimatic-mobile-frontend" (0.9.11)
> 15:23:31.191 [pimatic] Loading plugin: "pimatic-rflink" (0.6.0)
> 15:23:32.917 [pimatic-cron] the time is: Fri Oct 14 2016 15:23:32 GMT+0200 (CEST)
> 15:23:34.569 [pimatic] New device "RFSwitch"...
> 15:23:34.578 [pimatic] The device type RFLinkSwitch does not implement a destroy function
> 15:23:34.704 [pimatic-rflink] Opening connection to RFLink device...
> 15:23:34.745 [pimatic-rflink] Connection to RFLink device opened
> 15:23:35.203 [pimatic-mobile-frontend] packing static assets
> 15:23:36.008 [pimatic-mobile-frontend] packing static assets finished
> 15:23:36.026 [pimatic-mobile-frontend] rendering html
> 15:23:49.292 [pimatic-mobile-frontend] rendering html finished
> 15:23:49.418 [pimatic] Listening for HTTP-request on port 81...
> 15:23:49.465 [pimatic-rflink] No ready message received within connection timeout, reboot device
> 15:23:49.473 [pimatic-rflink] send: "10;REBOOT;"
> 15:23:49.492 [pimatic-rflink] data: "20;00;Nodo RadioFrequencyLink - RFLink Gateway V1.1 -Nanokatz R34;"
> 15:23:49.501 [pimatic-rflink] Received welcome message from RFLink
> 15:23:49.510 [pimatic-rflink] Connected to rflink device.
> 15:23:49.514 [pimatic-rflink] Enabling RFDEBUG...
> 15:23:49.575 [pimatic-rflink] send: "10;RFDEBUG=ON;"
> 15:23:49.592 [pimatic-rflink] data: "20;00;Nodo RadioFrequencyLink - RFLink Gateway V1.1 -Nanokatz R34;"
> 15:23:49.636 [pimatic-rflink] data: "20;01;RFDEBUG=ON;"
> 15:23:49.709 [pimatic] A uncaught exception occured: TypeError: resolver is not a function
> 15:23:49.709 [pimatic]>    at Board._handleAcknowledge (/mnt/elements/pimatic-app/node_modules/pimatic-rflink/lib/board.coffee:183:5)
> 15:23:49.709 [pimatic]>    at Board._processData (/mnt/elements/pimatic-app/node_modules/pimatic-rflink/lib/board.coffee:128:38)
> 15:23:49.709 [pimatic]>    at SerialPortDriver.<anonymous> (/mnt/elements/pimatic-app/node_modules/pimatic-rflink/lib/board.coffee:43:8)
> 15:23:49.709 [pimatic]>    at emitOne (events.js:77:13)
> 15:23:49.709 [pimatic]>    at SerialPortDriver.emit (events.js:169:7)
> 15:23:49.709 [pimatic]>    at SerialPort.<anonymous> (/mnt/elements/pimatic-app/node_modules/pimatic-rflink/lib/serialport.coffee:37:8)
> 15:23:49.709 [pimatic]>    at emitOne (events.js:77:13)
> 15:23:49.709 [pimatic]>    at SerialPort.emit (events.js:169:7)
> 15:23:49.709 [pimatic]>    at /mnt/elements/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/parsers.js:25:17
> 15:23:49.709 [pimatic]>    at Array.forEach (native)
> 15:23:49.709 [pimatic]>    at Object.parser (/mnt/elements/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/parsers.js:24:13)
> 15:23:49.709 [pimatic]>    at Object.SerialPort.opts.dataCallback (/mnt/elements/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/serialport.js:181:12)
> 15:23:49.709 [pimatic]>    at SerialPortFactory.SerialPort._emitData (/mnt/elements/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/serialport.js:396:20)
> 15:23:49.709 [pimatic]>    at afterRead (/mnt/elements/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/serialport.js:372:20)
> 15:23:49.709 [pimatic]>    at /mnt/elements/pimatic-app/node_modules/pimatic-rflink/node_modules/serialport/serialport.js:388:9
> 15:23:49.709 [pimatic]>    at FSReqWrap.wrapper [as oncomplete](fs.js:576:17)
> 15:23:49.709 [pimatic]> This is most probably a bug in pimatic or in a module, please report it!
> 15:23:49.721 [pimatic] shutting pimatic down...
> /mnt/elements/pimatic-app/node_modules/pimatic/startup.coffee:98
>           return framework.destroy().then(function() {
>                  ^

ReferenceError: framework is not defined
    at process.uncaughtException (/mnt/elements/pimatic-app/node_modules/pimatic/startup.coffee:85:9)
    at emitOne (events.js:77:13)
    at process.emit (events.js:169:7)
    at process._fatalException (node.js:224:26)

**Comment by nanokatz** [comment]
I found a workaround to avoid it crashing on me. opening the RFLink with minicom until after Pimatic is listening on HTTP will avoid this and let it connect and go on it's way

**Comment by sirchia** [comment]
Sorry for the late reply, but I just recently returned from vacation and broke my hand one day later :( 

I'll look into this issue as soon as I've recovered a bit. It doesn't sound too hard to solve and fortunately you've found a workaround, although I can imagine it's a bit inconvenient.

**Comment by NemoN** [comment]
> Sorry for the late reply, but I just recently returned from vacation and broke my hand one day later :(

Get well soon!

**Comment by ErikCoen** [comment]
When will be a new update to fix this issue?

**Comment by sirchia** [comment]
Issue is caused by the RFLink (just) not providing its welcome message within 10 seconds after initialisation of the serial connection (which is weird as mine responds in about 3 seconds). The plugin responds with a reboot of the device which apparently receives that request just after it sent the message, resulting in a second message after its restart.

Incorrect handling of the second (unexpected) welcome message caused the problem. This should be fixed in the new released version 0.7.0 of the plugin.

The reboot that is triggered due to timeout on the welcome message may however be completely prevented by providing the "connectionTimeout" option on the rflnk plugin configuration with a value greater than the default (which is 10000 -> 10 seconds). You can for example try 20000. You can know if that works by checking the debug log: you should not see the log of the timeout and reboot.

**Comment by ErikCoen** [comment]
This update and new option solve the issue for ever. Thanks!
